src/debuggability.js

```diff
@@ -161,6 +163,46 @@ Promise.prototype._propagateFrom = function (parent, flags) {
     USE(flags);
 };
 
+Promise.monitor = function () {
+    // Pending promises are stored in array for better performance: number
+    // of simultaneously pending promises is assumed to be relatively small,
+    // thus storing promises in the map and calculating hash for each promise
+    // is more expensive then iterating trough pending promises array when
+    // each promise is finished
+    Promise.prototype._pendingPromises = [];
```

Comment:
I would recommend using `Promise.monitor.pendingPromises` (if you want a public property) or just a private `var`.